src/createServer.js

```
@@ -1,13 +1,170 @@
-'use strict';
-
-// const express = require('express');
+const express = require('express');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
+  const app = express();
+
+  app.use(express.json());
+
+  let users = [];
+  let expenses = [];
+  let userIdCount = 1;
+  let expenseId = 1;
+
+  app.get('/users', (req, res) => {
+    return res.json(users);
+  });
+
+  app.get('/expenses', (req, res) => {
+    const { userId, categories, from, to } = req.query;
+    let filtered = [...expenses];
+
+    if (userId) {
+      filtered = filtered.filter((x) => x.userId === +userId);
+    }
+
+    if (categories) {
+      filtered = filtered.filter((x) => x.category === categories);
+    }
+
+    const fromDate = new Date(from);
+    const toDate = new Date(to);
+
+    if (from) {
+      filtered = filtered.filter((e) => fromDate <= new Date(e.spentAt));
+    }
+
+    if (to) {
+      filtered = filtered.filter((e) => new Date(e.spentAt) <= toDate);
+    }
```

GET /expenses uses truthy checks for query params (`if (userId)`, `if (categories)`, `if (from)`, `if (to)`). Truthy checks will skip valid but falsy values (for example `'0'` or `0` if such values are meaningful). Use explicit null/undefined checks (`if (userId != null)`, etc.) and coerce types consistently before comparing (e.g. `const uid = Number(userId); filtered = filtered.filter(x => x.userId === uid)`). This will make filtering deterministic and avoid silently ignoring queries.